---
title: "Biology"
author: "Nadine Bestard"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      dev = c('png', 'pdf'), 
                      warning = FALSE, message = FALSE)
```

```{r message=FALSE, warning=FALSE}
library(scater)
library(here)
library(ggplot2)
library(patchwork) # combine plots
library(MAST) # for DE
library(Seurat) # for MAST wrapper
library(VennDiagram)

# save figures
 save_path <- here("outs/Biology/", "/") # adding trailing / as filenames are pasted directly to path.
 dir.create(save_path)
# colours
blue_yellow <- ggsci::pal_jco("default")(2)
grey_red <- ggsci::pal_jco("default")(4)[3:4]
kelly_col <- unname(pals::kelly()[-c(1,2,3,6, 15)][c(2, 1 , 3:17)]) #exchange the yellow and purpule (so not all big clusters purpule) and delete similar colours
```

```{r}
  sce <- readRDS(here("outs", "singlecell-workflow",  "sce_anno_noNA.RDS"))

# subset to only have the main types in the violin plot
sce_sub <- sce[, sce$clusters_named %in% c("Astrocytes", "Microglia", "Oligodendrocyte", "Endothelial", "OPC")]
```

```{r markers,  fig.width=15, fig.path=save_path}
markers <- c("Cldn5","P2ry12","Mbp","Pdgfra")
plots <- lapply(markers, function(marker){
plotExpression(sce_sub, features = marker, x = "Sequencing", colour_by = "Sequencing", other_fields = "clusters_named" ) + facet_grid(~clusters_named) + scale_colour_manual(values = rev(blue_yellow)) + labs(y = marker) + 
    # make fonts bigger and delete usless labels
    theme(axis.text.x = element_blank(),
          axis.text.y = element_text(size = 20),
          axis.title.x = element_blank(),
          axis.title.y = element_text(size=30)) +
    # only add the celltypes in the first one.
    if(marker == "Cldn5"){
       theme(strip.background = element_blank(), strip.text = element_text(size=30))
    } else {
     theme(strip.background = element_blank(), strip.text = element_blank())
    }
      
      })
wrap_plots(plots, ncol=1)
```

# KO vs WT
First label the KO
```{r}
# the Samples with less than 5 microglia are KO - this is th3, 4, 5 and 7
table(sce$clusters_named, sce$Sample)["Microglia",]

sce$Genotype <- ifelse(grepl("TH5|TH7|TH3|TH4", sce$Sample), "KO", "WT")
table(sce$clusters_named, sce$Genotype)["Microglia",]
```

```{r}
# We will localise the DE to only 1 celltype. 
sce_oligo_mgi <- sce[,sce$clusters_named == "Oligodendrocyte" & sce$Sequencing == "MGI"]

sce_oligo_illumina <-  sce[,sce$clusters_named == "Oligodendrocyte" & sce$Sequencing == "Illumina"]

# DE
srt_oligo_mgi <- as.Seurat(sce_oligo_mgi, data = NULL, counts = "counts")
Idents(srt_oligo_mgi) <- srt_oligo_mgi$Genotype
srt_oligo_illumina <- as.Seurat(sce_oligo_illumina, data = NULL, counts = "counts")
Idents(srt_oligo_illumina) <- srt_oligo_illumina$Genotype
markers_mgi <- FindMarkers(srt_oligo_mgi, ident.1= "KO", ident.2 = "WT", test.use = "MAST")
markers_illumina <- FindMarkers(srt_oligo_illumina, ident.1= "KO", ident.2 = "WT", test.use = "MAST")

# how many do not overlap
sum(!(rownames(markers_mgi)) %in% rownames(markers_illumina))
sum(!(rownames(markers_illumina)) %in% rownames(markers_mgi))

```
The genes that do not overlap have little difference in pct and small logFC. 
If we set stricter thresholds we see how only 3 genes do not overlap.
```{r}


# markers stricter
markers_strict_mgi <- FindMarkers(srt_oligo_mgi, ident.1= "KO", ident.2 = "WT", test.use = "MAST", logfc.threshold = 0.3 )
markers_strict_illumina <- FindMarkers(srt_oligo_illumina, ident.1= "KO", ident.2 = "WT", test.use = "MAST", logfc.threshold = 0.3)

sum(!(rownames(markers_strict_illumina)) %in% rownames(markers_strict_mgi))
markers_strict_illumina[(!(rownames(markers_strict_illumina)) %in% rownames(markers_strict_mgi)),]
dim(markers_strict_illumina)
dim(markers_strict_mgi)


```

```{r}
#the p-value has a greater spread in the common genes. 
markers_illumina %>% mutate(only= !(rownames(markers_illumina)) %in% rownames(markers_mgi)) %>% group_by(only) %>% summarise(mean(p_val_adj), mean(abs((avg_log2FC))), min(p_val_adj), max(p_val_adj))

markers_mgi %>% mutate(only= !(rownames(markers_mgi)) %in% rownames(markers_illumina)) %>% group_by(only) %>% summarise(mean(p_val_adj), mean(abs(avg_log2FC)))

```

```{r venngenes}
save_path <- here("outs", "singlecell-workflow")
venn.diagram(list(rownames(markers_mgi), rownames(markers_illumina)), 
             
             #aes
             col = blue_yellow,
             fill = c(alpha(blue_yellow[1]), alpha(blue_yellow[2])),
             #save
             height = 4800,
             width = 7000,
             resolution = 300,
             filename = here(save_path, "markers_KO_WT.png"),
             disable.logging=TRUE,
             #text nums
             cex=4,
             fontfamily="sans",
             #text categories
             category.names = c("MGI" , "Illumina"),
             cat.cex = 4,
             cat.fontfamily="sans",
             cat.default.pos="outer",
             cat.dist = c(0.055, 0.055),
             #title
             main="Genes DE between KO and WT",
             main.fontfamily = "sans",
             main.cex=5
             )
```

```{r venngenes_strict}
venn.diagram(list(rownames(markers_strict_mgi), rownames(markers_strict_illumina)), 
             
             #aes
             col = blue_yellow,
             fill = c(alpha(blue_yellow[1]), alpha(blue_yellow[2])),
             #save
             height = 4800,
             width = 7000,
             resolution = 300,
             filename = here(save_path, "markers_KO_WT_filtered.png"),
             disable.logging=TRUE,
             #text nums
             cex=4,
             fontfamily="sans",
             #text categories
             category.names = c("MGI" , "Illumina"),
             cat.cex = 4,
             cat.fontfamily="sans",
             cat.default.pos="outer",
             cat.dist = c(0.055, 0.055),
             #title
             main="Genes DE between KO and WT",
             main.fontfamily = "sans",
             main.cex=5
             )
```

```{r umap_facet_genotype, fig.path=save_path}
plotReducedDim(sce, dimred="TSNE", other_fields = "Genotype", colour_by = "clusters_named") + facet_wrap(~Genotype) + scale_colour_manual(values = kelly_col)
```

```{r}
markers_mgi$gene <- rownames(markers_mgi)
markers_illumina$gene <- rownames(markers_illumina)

```
```{r}
EnhancedVolcano::EnhancedVolcano(markers_illumina, x = "avg_log2FC", y = "p_val_adj", lab=rownames(markers_illumina), FCcutoff=0.3)
```

```{r}
markers_illumina %>% 
  filter(!(gene %in% rownames(markers_mgi))) %>% 
           EnhancedVolcano::EnhancedVolcano(markers_illumina, x = "avg_log2FC", y = "p_val_adj", lab="gene", FCcutoff=0.2)
```
```{r}
markers <- full_join(markers_illumina, markers_mgi,suffix = c("_illumina", "_mgi"), by = "gene")
markers <- markers %>% mutate(only_illumina = !(gene %in% rownames(markers_mgi)),
                             only_mgi = !(gene %in% rownames(markers_illumina))
                             )
markers$sequencing <-  ifelse(markers$only_illumina, "only_illumina", 
                        ifelse(markers$only_mgi, "only_mgi", "both")) 
markers <- markers %>% rowwise() %>% mutate(avg_log2FC_avg = mean(c(avg_log2FC_illumina, avg_log2FC_mgi), na.rm = TRUE))

markers %>% ggplot2::ggplot(aes(y = avg_log2FC_illumina)) + geom_boxplot() 
markers %>% ggplot2::ggplot(aes(y = avg_log2FC_mgi)) + geom_boxplot() 
markers %>% 
  filter(!(gene %in% rownames(markers_mgi))) %>%  ggplot2::ggplot(aes(y = avg_log2FC)) + geom_boxplot() 
markers_mgi %>% 
  filter(!(gene %in% rownames(markers_illumina))) %>%  ggplot2::ggplot(aes(y = abs(avg_log2FC), x = dummy_variable)) + geom_boxplot() + ylim (c(0,0.5)) +
  geom_jitter(color = "#0073C2FF"  ) +   scale_fill_manual(values = "#0073C2FF") 

```

```{r}
markers %>% ggplot2::ggplot(aes(y = abs(avg_log2FC_avg), x = sequencing)) + geom_boxplot(outlier.alpha = 0) + ylim (c(0,2.5))  +
  geom_jitter(aes(color = sequencing ) ) + scale_color_manual(values = c("darkgreen","#EFC000FF" , "#0073C2FF" ) ) +
  labs(x = NULL,
       y = "Abs(log2FC)") +
  geom_text(label = "C4b", x= "both", y = 2.5) +
  geom_text(label = "Serpina3n", x= "both", y = 1.8) +
  theme_classic()
```
```{r}
ggsave(here("outs/Biology/logFC_small.pdf"), width = 5, height = 4 )
ggsave(here("outs/Biology/logFC.png"), width = 5, height = 4 )
```

## pvalues
less visually clear
```{r}
markers %>% ggplot2::ggplot(aes(y = (p_val_adj_avg), x = sequencing)) + geom_boxplot(outlier.alpha = 0) + ylim (c(0,2.5))  +
  geom_jitter(aes(color = sequencing ) ) + scale_color_manual(values = c("darkgreen","#EFC000FF" , "#0073C2FF" ) ) +
  labs(x = NULL,
       y = "pval") +
  scale_y_log10()+
 # geom_text(label = "C4b", x= "both", y = 2.5) +
 # geom_text(label = "Serpina3n", x= "both", y = 1.8) +
  theme_classic()
```

```{r}
markers %>% filter(avg_log2FC_avg > 0.3) %>% count(sequencing)
markers %>% filter(sequencing != "both") %>% .$avg_log2FC_avg  %>% summary(
```

```{r}
sce_oligo <- sce[,sce$clusters_named == "Oligodendrocyte"]
```


```{r}
# Plot Serpin3n
sce_oligo$random_order <- sample(ncol(sce_oligo))
plotTSNE(sce_oligo[,sce_oligo$random_order], colour_by = "C4b", other_fields = "Sequencing") + lims(x = c(-15, 10), y = c(-30, 0)) +  #scale_colour_manual(values = c(kelly_col[1:2])) +
  facet_wrap(~Sequencing)
ggsave(here("outs/Biology/tsne_c4b.pdf"), width = 4, height = 3)
ggsave(here("outs/Biology/tsne_c4b.png"), width = 4, height = 3)

plotTSNE(sce_oligo[,sce_oligo$random_order], colour_by = "Serpina3n", other_fields = "Sequencing") + lims(x = c(-15, 10), y = c(-30, 0)) +  #scale_colour_manual(values = c(kelly_col[1:2])) +
  facet_wrap(~Sequencing)
ggsave(here("outs/Biology/tsne_Serpina3n.pdf"), width = 4, height = 3)
ggsave(here("outs/Biology/tsne_Serpina3n.png"), width = 4, height = 3)
```
```{r}
# Plot DE
sce_oligo_mgi$random_order <- sample(ncol(sce_oligo_mgi))
plotTSNE(sce_oligo_mgi[,sce_oligo_mgi$random_order], colour_by = "Genotype") + lims(x = c(-15, 10), y = c(-30, 0)) +  scale_colour_manual(values = c(kelly_col[1:2])) +
  scale_alpha(1)
ggsave(here("outs/Biology/tsne_ko_wt.pdf"), width = 3, height = 3)
ggsave(here("outs/Biology/tsne_ko_wt.png"), width = 3, height = 3)
```


# tables
```{r}
markers_mgi %>% arrange(p_val_adj) %>% slice(1:11) %>% write.csv(here("outs/Biology/top_pval_mgi.csv"))
markers_illumina %>% arrange(p_val_adj) %>% slice(1:11) %>% write.csv(here("outs/Biology/top_pval_illumina.csv"))

markers_mgi %>% arrange(desc(abs(avg_log2FC))) %>% slice(1:11) %>% write.csv(here("outs/Biology/top_logFC_mgi.csv"))

markers_illumina %>% arrange(desc(abs(avg_log2FC))) %>% slice(1:11) %>% write.csv(here("outs/Biology/top_logFC_illumina.csv"))
```

