---
title: "MGI vs Illumina"
author: "Nadine Bestard"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r}
library(here)
library(dplyr)
library(ggplot2)
```

# ALL

## FastQC

```{r}
fastqc <- read.delim(here("outs/fastqc/downsampled/multiqc_data/multiqc_fastqc.txt"))
fastqc_general <- read.delim(here("outs/fastqc/downsampled/multiqc_data/multiqc_general_stats.txt"))

fastqc_r2 <- fastqc_general %>% 
  mutate(Sequencing = ifelse(grepl("MGI", fastqc$Sample), "MGI", "Illumina")) %>% 
  #remove the R1
  filter(FastQC_mqc.generalstats.fastqc.avg_sequence_length == 90) %>%
  #remove the RT
  filter(FastQC_mqc.generalstats.fastqc.total_sequences > 150000000 & 
           Sample != ("MGI_RT1_S1_L001_R2_001") &
           Sample != ("MGI_RT2_S1_L001_R2_001")) # %>% 
 
  fastqc_r2 %>%  group_by(Sequencing) %>%
  summarise(dups=mean(FastQC_mqc.generalstats.fastqc.percent_duplicates)) %>%
  # plot the duplicated seqs
  ggplot(aes(x=Sequencing, y=dups)) + geom_bar(stat = "identity")

#total_deduplicated_percentage
  MGI_dups <- fastqc_r2 %>% 
    filter(Sequencing == "MGI") %>%
    select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  illumina_dups <- fastqc_r2 %>% filter(Sequencing == "Illumina") %>% select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  t.test(MGI_dups, illumina_dups, paried = TRUE )
  
  # only moused
  
  #total_deduplicated_percentage
  MGI_dups <- fastqc_r2 %>% 
    filter(Sequencing == "MGI") %>% 
    select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  illumina_dups <- fastqc_r2 %>% filter(Sequencing == "Illumina") %>%  select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  t.test(MGI_dups, illumina_dups, paried = TRUE )
  
```
```{r}
PairedData::plot(PairedData::paired(MGI_dups, illumina_dups), type = "profile", ylim= c(0,100), cols = c("darkblue", "darkorange")) # to do, replot with ggpubr
ggpubr::ggpaired(fastqc_r2, x = "Sequencing", y = "FastQC_mqc.generalstats.fastqc.percent_duplicates" , fill = "Sequencing", palette = "jco") #+ ylim(c(0,100))
# "#0073C2FF" "#EFC000FF" 
```
With RT not possible paired data, cause illumina is in 2 lanes. ( I could have merged them, but I haven't)

```{r, eval=FALSE}
fastqc_r2_rt <- fastqc_general %>% 
  mutate(Sequencing = ifelse(grepl("MGI", fastqc$Sample), "MGI", "Illumina")) %>% 
  #remove the R1
  filter(FastQC_mqc.generalstats.fastqc.avg_sequence_length == 90) %>%
  #remove the RT
  filter(FastQC_mqc.generalstats.fastqc.total_sequences < 290000000 | 
           Sample == ("MGI_RT1_S1_L001_R2_001") |
           Sample == ("MGI_RT2_S1_L001_R2_001")) # %>% 
 
  fastqc_r2 %>%  group_by(Sequencing) %>%
  summarise(dups=mean(FastQC_mqc.generalstats.fastqc.percent_duplicates)) %>%
  # plot the duplicated seqs
  ggplot(aes(x=Sequencing, y=dups)) + geom_bar(stat = "identity")

fastqc_r2_rt
  
  
```

![sequencequalityscores][../outs/fastqcplots/fastqc_per_sequence_quality_scores_plot_orange_blue]


## Cell Ranger

- plot umis, mapped, cells, avg reads/cells 
```{r}
# all interesting info was already in the other file
#cellranger_general <- read.delim(here("outs/cellranger/downsampled/multiqc_data/multiqc_general_stats.txt"))
#colnames(cellranger_general) <- stringr::str_replace(colnames(cellranger_general),"Cell.Ranger.Count_mqc.generalstats.cell_ranger_count.Count_", "")

cellranger <- read.delim(here("outs/cellranger/downsampled/multiqc_data_all/multiqc_cellranger_count.txt"))

#cellranger <- merge(cellranger_general, cellranger, by= "Sample")

# illumina reads TH7 and TH8 are swapped

cellranger[c(18,19),2:21] <- cellranger[c(19,18),2:21]


cellranger_general <- cellranger %>% select(Sample, estimated.cells, reads.in.cells, avg.reads.cell, saturation, genes.detected) %>% 
  mutate(Sequencing = ifelse(grepl("MGI", cellranger$Sample), "MGI", "Illumina"))

cellranger_seq <- cellranger %>% select(Sample, 
reads.mapped,
confident.intergenic, confident.intronic, confident.exonic, reads.antisense) %>% 
  mutate(Sequencing = ifelse(grepl("MGI", cellranger$Sample), "MGI", "Illumina"))


for(variable in colnames(cellranger_seq)){
  plot(ggpubr::ggpaired(cellranger_seq, x = "Sequencing", y = variable , fill = "Sequencing", palette = "jco")  + ylab(variable))
}

for(variable in colnames(cellranger_general)){
  plot(ggpubr::ggpaired(cellranger_general, x = "Sequencing", y = variable , fill = "Sequencing", palette = "jco")  + ylab(variable))
}

```
```{r}
#t-test
for(variable in colnames(cellranger_general)){
  if(variable != "Sample" & variable != "Sequencing" ){
 MGI <- cellranger_general %>% 
    filter(Sequencing == "MGI") %>% 
    select(variable)
  
  illumina <- cellranger_general %>% filter(Sequencing == "Illumina") %>%  select(variable)
  
  print(variable)
 print( t.test(MGI, illumina
         , paried = TRUE ))
  }
}

for(variable in colnames(cellranger_seq)){
  if(variable != "Sample" & variable != "Sequencing" ){
 MGI <- cellranger_seq %>% 
    filter(Sequencing == "MGI") %>% 
    select(variable)
  
  illumina <- cellranger_seq %>% filter(Sequencing == "Illumina") %>%  select(variable)
  
  print(variable)
 print( t.test(MGI, illumina
         , paried = TRUE ))
  }
}
```


# MOUSE

## FastQC

```{r}
fastqc <- read.delim(here("outs/fastqc/downsampled/multiqc_data/multiqc_fastqc.txt"))
fastqc_general <- read.delim(here("outs/fastqc/downsampled/multiqc_data/multiqc_general_stats.txt"))

fastqc_r2 <- fastqc_general %>% 
  mutate(Sequencing = ifelse(grepl("MGI", fastqc$Sample), "MGI", "Illumina")) %>% 
  #remove the R1
  filter(FastQC_mqc.generalstats.fastqc.avg_sequence_length == 90) %>%
  #remove the RT and the ZB
  filter(FastQC_mqc.generalstats.fastqc.total_sequences > 150000000 & 
           grepl("TH", Sample)) # %>% 
  
  fastqc_r2 %>%  group_by(Sequencing) %>%
  summarise(dups=mean(FastQC_mqc.generalstats.fastqc.percent_duplicates)) %>%
  # plot the duplicated seqs
  ggplot(aes(x=Sequencing, y=dups)) + geom_bar(stat = "identity")

#total_deduplicated_percentage
  MGI_dups <- fastqc_r2 %>% 
    filter(Sequencing == "MGI") %>%
    select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  illumina_dups <- fastqc_r2 %>% filter(Sequencing == "Illumina") %>% select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  t.test(MGI_dups, illumina_dups, paried = TRUE )
  

  
```
```{r}
PairedData::plot(PairedData::paired(MGI_dups, illumina_dups), type = "profile", ylim= c(0,100), cols = c("darkblue", "darkorange")) # to do, replot with ggpubr
ggpubr::ggpaired(fastqc_r2, x = "Sequencing", y = "FastQC_mqc.generalstats.fastqc.percent_duplicates" , fill = "Sequencing", palette = "jco") #+ ylim(c(0,100))
# "#0073C2FF" "#EFC000FF" 
```


![sequencequalityscores][../outs/fastqcplots/fastqc_per_sequence_quality_scores_plot_orange_blue]


## Cell Ranger

- plot umis, mapped, cells, avg reads/cells 
```{r}
# all interesting info was already in the other file
#cellranger_general <- read.delim(here("outs/cellranger/downsampled/multiqc_data/multiqc_general_stats.txt"))
#colnames(cellranger_general) <- stringr::str_replace(colnames(cellranger_general),"Cell.Ranger.Count_mqc.generalstats.cell_ranger_count.Count_", "")

cellranger <- read.delim(here("outs/cellranger/downsampled/multiqc_data_all/multiqc_cellranger_count.txt"))

#cellranger <- merge(cellranger_general, cellranger, by= "Sample")

# illumina reads TH7 and TH8 are swapped

cellranger[c(18,19),2:21] <- cellranger[c(19,18),2:21]


cellranger_general <- cellranger %>% 
  filter(grepl("TH", Sample)) %>%  # only mouse
  select(Sample, estimated.cells, reads.in.cells, avg.reads.cell, saturation, genes.detected) %>% 
  mutate(Sequencing = ifelse(grepl("MGI", Sample), "MGI", "Illumina"))

cellranger_seq <- cellranger %>%
  filter(grepl("TH", Sample)) %>%  # only mouse
  select(Sample, 
reads.mapped,
confident.intergenic, confident.intronic, confident.exonic, reads.antisense) %>% 
  mutate(Sequencing = ifelse(grepl("MGI", Sample), "MGI", "Illumina"))



for(variable in colnames(cellranger_seq)){
  plot(ggpubr::ggpaired(cellranger_seq, x = "Sequencing", y = variable , fill = "Sequencing", palette = "jco")  
       + ylab(variable) +
         ylim(c(0, max(cellranger_seq[[variable]])))
       )
}

for(variable in colnames(cellranger_general)){
  plot(ggpubr::ggpaired(cellranger_general, x = "Sequencing", y = variable , fill = "Sequencing", palette = "jco")  + ylab(variable))
}

```
```{r}
#t-test
for(variable in colnames(cellranger_general)){
  if(variable != "Sample" & variable != "Sequencing" ){
 MGI <- cellranger_general %>% 
    filter(Sequencing == "MGI") %>% 
    select(variable)
  
  illumina <- cellranger_general %>% filter(Sequencing == "Illumina") %>%  select(variable)
  
  print(variable)
 print( t.test(MGI, illumina
         , paried = TRUE ))
  }
}

for(variable in colnames(cellranger_seq)){
  if(variable != "Sample" & variable != "Sequencing" ){
 MGI <- cellranger_seq %>% 
    filter(Sequencing == "MGI") %>% 
    select(variable)
  
  illumina <- cellranger_seq %>% filter(Sequencing == "Illumina") %>%  select(variable)
  
  print(variable)
 print( t.test(MGI, illumina
         , paried = TRUE ))
  }
}
```

## save the plots I'll use in the paper
```{r}
blue_yellow <- ggsci::pal_jco("default")(2)
ggplot(cellranger_seq, aes(x = Sequencing, y = reads.mapped , fill = Sequencing, colour = Sequencing))  +
  geom_point(shape = 21) + #colour = "black", lwd = 0.001
  ylab("% Reads Mapped") +
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0, max(cellranger_seq$reads.mapped)))
       
```
```{r}
blue_yellow <- ggsci::pal_jco("default")(2)
ggplot(cellranger_seq, aes(x = Sequencing, y = reads.mapped , fill = Sequencing))  +
  ylab("% Reads Mapped") +
  geom_boxplot( lwd = 0.001) + 
  geom_point(pch=21, stroke=0.01) +
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0, 100)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/mapping_boxplot.pdf"), width = 5, height = 3) 
       
```
```{r}
ggplot(fastqc_r2, aes(x = Sequencing, y = FastQC_mqc.generalstats.fastqc.percent_duplicates, fill = Sequencing))  +
  ylab("% Duplicated reads") +
geom_boxplot( lwd = 0.01, outlier.shape = 21, outlier.fill = blue_yellow[2], outlier.stroke = 0.01) +
  geom_jitter(pch=21, width=0, stroke = 0.01)+
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0,100)) +
  theme_classic()

ggsave(here("outs/fastqc/boxplots/duplicatedreads_boxplot.pdf"), width = 5, height = 3) 
```

```{r}
ggplot(cellranger_general, aes(x = Sequencing, y = saturation, fill = Sequencing))  +
  ylab("Saturation") +
  geom_boxplot( lwd = 0.01, outlier.shape = 21, outlier.fill = blue_yellow[2], outlier.stroke = 0.01) +
  geom_jitter(pch=21, width=0, stroke = 0.01)+
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0, 100)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/saturation_boxplot.pdf"), width = 5, height = 3) 
```
```{r}
ggplot(cellranger_seq, aes(x = Sequencing, y = confident.intronic, fill = Sequencing))  +
  ylab("% Reads mapped to intronic region") +

geom_boxplot( lwd = 0.001) + 
  geom_point(pch=21, stroke=0.01) +
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0, 100)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/intronic_boxplot.pdf"), width = 5, height = 3) 
```
```{r}
ggplot(cellranger_seq, aes(x = Sequencing, y = confident.exonic, fill = Sequencing))  +
  ylab("% Reads mapped to exonic region") +

geom_boxplot( lwd = 0.001) + 
  geom_point(pch=21, stroke=0.01) +
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0, 100)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/exonic_boxplot.pdf"), width = 5, height = 3) 
```
```{r}
ggplot(cellranger_seq, aes(x = Sequencing, y = reads.antisense, fill = Sequencing))  +
  ylab("% Reads mapped to antisense region") +

geom_boxplot( lwd = 0.001) + 
  geom_point(pch=21, stroke=0.01) +
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0, 100)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/antisense_boxplot.pdf"), width = 5, height = 3) 
```

```{r}
ggplot(cellranger_general, aes(x = Sequencing, y = reads.in.cells , fill = Sequencing))  +
  ylab("% Reads in Cells") +
  geom_boxplot( lwd = 0.01, outlier.shape = 21, outlier.fill = blue_yellow[2], outlier.stroke = 0.01) +
  geom_jitter(pch=21, width=0, stroke = 0.01)+
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
         ylim(c(0, 100)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/reads_in_cells_boxplot.pdf"), width = 5, height = 3)
```

```{r}
ggplot(cellranger_general, aes(x = Sequencing, y = estimated.cells , fill = Sequencing))  +
  ylab("Estimated cell number") +
  geom_boxplot( lwd = 0.01, outlier.shape = 21, outlier.fill = blue_yellow[2], outlier.stroke = 0.01) +
  geom_jitter(pch=21, width=0, stroke = 0.01)+
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/cells_boxplot.pdf"), width = 5, height = 3)
```

```{r}
ggplot(cellranger_general, aes(x = Sequencing, y = genes.detected , fill = Sequencing))  +
  ylab("Number of Genes") +
  geom_boxplot( lwd = 0.01, outlier.shape = 21, outlier.fill = blue_yellow[2], outlier.stroke = 0.01) +
  geom_jitter(pch=21, width=0, stroke = 0.01)+
  scale_fill_manual(values = rev(blue_yellow)) +
   scale_colour_manual(values = rev(blue_yellow)) +
  theme_classic()
ggsave(here("outs/fastqc/boxplots/genes_boxplot.pdf"), width = 5, height = 3)
```
# ZEBRAFISH

## FastQC

```{r}
fastqc <- read.delim(here("outs/fastqc/downsampled/multiqc_data/multiqc_fastqc.txt"))
fastqc_general <- read.delim(here("outs/fastqc/downsampled/multiqc_data/multiqc_general_stats.txt"))

fastqc_r2 <- fastqc_general %>% 
  mutate(Sequencing = ifelse(grepl("MGI", fastqc$Sample), "MGI", "Illumina")) %>% 
  #remove the R1
  filter(FastQC_mqc.generalstats.fastqc.avg_sequence_length == 90) %>%
  #remove the RT and the ZB
  filter(FastQC_mqc.generalstats.fastqc.total_sequences > 150000000 & 
           grepl("ZB", Sample)) # %>% 
  
  fastqc_r2 %>%  group_by(Sequencing) %>%
  summarise(dups=mean(FastQC_mqc.generalstats.fastqc.percent_duplicates)) %>%
  # plot the duplicated seqs
  ggplot(aes(x=Sequencing, y=dups)) + geom_bar(stat = "identity")

#total_deduplicated_percentage
  MGI_dups <- fastqc_r2 %>% 
    filter(Sequencing == "MGI") %>%
    select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  illumina_dups <- fastqc_r2 %>% filter(Sequencing == "Illumina") %>% select(FastQC_mqc.generalstats.fastqc.percent_duplicates)
  
  t.test(MGI_dups, illumina_dups, paried = TRUE )
  

  
```
```{r}
PairedData::plot(PairedData::paired(MGI_dups, illumina_dups), type = "profile", ylim= c(0,100), cols = c("darkblue", "darkorange")) # to do, replot with ggpubr
ggpubr::ggpaired(fastqc_r2, x = "Sequencing", y = "FastQC_mqc.generalstats.fastqc.percent_duplicates" , fill = "Sequencing", palette = "jco") #+ ylim(c(0,100))
# "#0073C2FF" "#EFC000FF" 
```

## Cell Ranger

- plot umis, mapped, cells, avg reads/cells 
```{r}
# all interesting info was already in the other file
#cellranger_general <- read.delim(here("outs/cellranger/downsampled/multiqc_data/multiqc_general_stats.txt"))
#colnames(cellranger_general) <- stringr::str_replace(colnames(cellranger_general),"Cell.Ranger.Count_mqc.generalstats.cell_ranger_count.Count_", "")

cellranger <- read.delim(here("outs/cellranger/downsampled/multiqc_data_all/multiqc_cellranger_count.txt"))

#cellranger <- merge(cellranger_general, cellranger, by= "Sample")

# illumina reads TH7 and TH8 are swapped

cellranger[c(18,19),2:21] <- cellranger[c(19,18),2:21]


cellranger_general <- cellranger %>% 
  filter(grepl("ZB", Sample)) %>%  # only fish
  select(Sample, estimated.cells, reads.in.cells, avg.reads.cell, saturation, genes.detected) %>% 
  mutate(Sequencing = ifelse(grepl("MGI", Sample), "MGI", "Illumina"))

cellranger_seq <- cellranger %>%
  filter(grepl("ZB", Sample)) %>%  # only mouse
  select(Sample, 
reads.mapped,
confident.intergenic, confident.intronic, confident.exonic, reads.antisense) %>% 
  mutate(Sequencing = ifelse(grepl("MGI", Sample), "MGI", "Illumina"))


for(variable in colnames(cellranger_seq)){
  plot(ggpubr::ggpaired(cellranger_seq, x = "Sequencing", y = variable , fill = "Sequencing", palette = "jco")  + ylab(variable))
}

for(variable in colnames(cellranger_general)){
  plot(ggpubr::ggpaired(cellranger_general, x = "Sequencing", y = variable , fill = "Sequencing", palette = "jco")  + ylab(variable))
}

```
```{r}
#t-test
for(variable in colnames(cellranger_general)){
  if(variable != "Sample" & variable != "Sequencing" ){
 MGI <- cellranger_general %>% 
    filter(Sequencing == "MGI") %>% 
    select(variable)
  
  illumina <- cellranger_general %>% filter(Sequencing == "Illumina") %>%  select(variable)
  
  print(variable)
 print( t.test(MGI, illumina
         , paried = TRUE ))
  }
}

for(variable in colnames(cellranger_seq)){
  if(variable != "Sample" & variable != "Sequencing" ){
 MGI <- cellranger_seq %>% 
    filter(Sequencing == "MGI") %>% 
    select(variable)
  
  illumina <- cellranger_seq %>% filter(Sequencing == "Illumina") %>%  select(variable)
  
  print(variable)
 print( t.test(MGI, illumina
         , paried = TRUE ))
  }
}
```
