---
title: "Thalamus David m. data MGI vs Illumina"
author: "Nadine Bestard"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, 
                      dev = c('png', 'pdf'), 
                      warning = FALSE, message = FALSE)
```

# Libraries
```{r message=FALSE, warning=FALSE}
library(DropletUtils)
library(here)
library(ggplot2)
library(dplyr) #case_when

# save figures
 save_path <- here("outs/singlecell-workflow", "/") # adding trailing / as filenames are pasted directly to path.
 dir.create(save_path)
```
```{r}
# colours
blue_yellow <- ggsci::pal_jco("default")(2)
grey_red <- ggsci::pal_jco("default")(4)[3:4]
kelly_col <- unname(pals::kelly()[-c(1,2,3,6, 15)][c(2, 1 , 3:17)]) #exchange the yellow and purpule (so not all big clusters purpule) and delete similar colours
# scales::show_col(ggsci::pal_jco("default")(4))
```


# Load datasets
```{r}
#thalamus
sample_names_th_illumina <-dir(here("outs/cellranger/downsampled"), pattern = "^[T]") # extract TH and ZB
sample_names_th_mgi <-dir(here("outs/cellranger/downsampled"), pattern = "^MGI_[T]") # extract TH and ZB
#sampels
samples_th_illumina <- here(paste0("outs/cellranger/downsampled/", sample_names_th_illumina, "/outs/filtered_feature_bc_matrix/"))
samples_th_mgi <- here(paste0("outs/cellranger/downsampled/", sample_names_th_mgi, "/outs/filtered_feature_bc_matrix/"))

#into sce
sce_th_mgi <- read10xCounts(samples_th_mgi, sample.names = sample_names_th_mgi, col.names = TRUE)
sce_th_illumina <- read10xCounts(samples_th_illumina, sample.names = sample_names_th_illumina, col.names = TRUE)

# Mark the cells
cells_only_in_illumina <- readRDS(here("outs/correlations/cells_only_in_illumina.RDS"))
sce_th_illumina$Only_in_one_sequencing_tecnology <- ifelse(colnames(sce_th_illumina) %in% cells_only_in_illumina, "Only_Illumina", "both")
sce_th_illumina$Barcode_unique <- paste0(colnames(sce_th_illumina), "_Illumina")
sce_th_illumina$Sequencing <- "Illumina"

cells_only_in_mgi <- readRDS(here("outs/correlations/cells_only_in_mgi.RDS"))
sce_th_mgi$Only_in_one_sequencing_tecnology <- ifelse(colnames(sce_th_mgi) %in% cells_only_in_mgi, "Only_MGI", "both")
sce_th_mgi$Barcode_unique <- paste0(colnames(sce_th_mgi), "_MGI")
sce_th_mgi$Sequencing <- "MGI"

# Merge both
sce <- cbind(sce_th_illumina, sce_th_mgi)
```

## Add QC info

```{r}
library(scater)
rownames(sce) <- uniquifyFeatureNames(rowData(sce)$ID, 
    rowData(sce)$Symbol)
is_mito <- grepl("^mt-", rownames(sce))
sce <- addPerCellQCMetrics(sce, subsets=list(Mt=is_mito))
```


# Quality control with linient filters


## Cell qc
I use here automatic filters that are quite linient. 

```{r}
qc <- perCellQCFilters(sce, sub.fields = "subsets_Mt_percent")
colData(sce) <- cbind(colData(sce), qc)
```


```{r}
plotColData(sce, y = "sum", colour_by = "discard", x = "Only_in_one_sequencing_tecnology") +
  scale_colour_manual(values = grey_red)
plotColData(sce, y = "subsets_Mt_percent", colour_by = "discard", x = "Only_in_one_sequencing_tecnology") +
  scale_colour_manual(values = grey_red)
```

The cells thatwere only detected in one of the sequencing tecnologies are low quality, and are discarded even with a linient filter.
Significantly more cells are discarded from the "only in illumina" and "only in MGI" group than from the rest of cells. 

```{r}
discardnum <- table(sce$Only_in_one_sequencing_tecnology, sce$discard)
colnames(discardnum) <- c("Keep", "Discard")
discardnum
# calculate proportion of discarded cells in each group
discardnum[,2]/(discardnum[,1]+discardnum[,2])*100 # the expected would be 17.34 for all.
#chisq tests contingency table tests, is the proportion between keep and discarded different than the expected if the groups had no effect on these proportions?
# The numbers are so big that anything would be significant, so I divided by 100, that is the order of magnitude we are interested in.
chisq.test(discardnum/100)
# this is the expected values if all were the same
chisq.test(discardnum)$expected
```

For the cells detected in both technologies, are we deleting more MGI or Illumina cells?
Maybe slightly more Illumina ones, but not significant
```{r}
plotColData(sce, y = "sum", colour_by = "discard", x = "Sequencing") +
  scale_colour_manual(values = grey_red) + ylab("UMI counts")
plotColData(sce, y = "detected", colour_by = "discard", x = "Sequencing")+
  scale_colour_manual(values = grey_red) + ylab("Detected genes")
plotColData(sce, y = "subsets_Mt_percent", colour_by = "discard", x = "Sequencing")+
  scale_colour_manual(values = grey_red) + ylab("Mitochondrial percentage")
```

```{r}
discardnum <- table(sce$Sequencing, sce$discard)
colnames(discardnum) <- c("Keep", "Discard")
discardnum
# calculate proportion of discarded cells in each group
discardnum[,2]/(discardnum[,1]+discardnum[,2])*100 # the expected would be 17.34 for all.
#chisq tests contingency table tests, is the proportion between keep and discarded different than the expected if the groups had no effect on these proportions?
chisq.test(discardnum/100)#, simulate.p.value = TRUE)
# this is the expected values if all proportions were the same
chisq.test(discardnum)$expected
```

# Quality control with true filters

## Cell QC
I use here the same thresholds applied when analysing the biological data, after multiple rounds of QC

```{r}
sce$filter <- sce$sum < 3000 | sce$detected < 750 | sce$subsets_Mt_percent > 6

```


```{r violin_only_one_seq, fig.path=save_path}
plotColData(sce, y = "sum", colour_by = "filter", x = "Only_in_one_sequencing_tecnology") +
  scale_colour_manual(values = grey_red)
plotColData(sce, y = "subsets_Mt_percent", colour_by = "filter", x = "Only_in_one_sequencing_tecnology") +
  scale_colour_manual(values = grey_red)
```

With stricter filters all the cells only detected with one technology are reomoved

```{r}
filternum <- table(sce$Only_in_one_sequencing_tecnology, sce$filter)
colnames(filternum) <- c("Keep", "filter")
filternum
# calculate proportion of filtered cells in each group
filternum[,2]/(filternum[,1]+filternum[,2])*100 # the expected would be 17.34 for all.
#chisq tests contingency table tests, is the proportion between keep and filtered different than the expected if the groups had no effect on these proportions?
chisq.test(filternum)
# this is the expected values if all were the same
chisq.test(filternum)$expected
```

For the cells detected in both technologies, are we deleting more MGI or Illumina cells?
Here the difference is even smaller between the two technologies
```{r violin_sequencing, fig.path=save_path}
plotColData(sce, y = "sum", colour_by = "filter", x = "Sequencing") +
  scale_colour_manual(values = grey_red) + ylab("UMI counts")
plotColData(sce, y = "detected", colour_by = "filter", x = "Sequencing")+
  scale_colour_manual(values = grey_red) + ylab("Detected genes")
plotColData(sce, y = "subsets_Mt_percent", colour_by = "filter", x = "Sequencing")+
  scale_colour_manual(values = grey_red) + ylab("Mitochondrial percentage")
```

```{r}
filternum <- table(sce$Sequencing, sce$filter)
colnames(filternum) <- c("Keep", "filter")
filternum
# calculate proportion of filtered cells in each group
filternum[,2]/(filternum[,1]+filternum[,2])*100 # the expected would be 17.34 for all.
#chisq tests contingency table tests, is the proportion between keep and filtered different than the expected if the groups had no effect on these proportions?
chisq.test(filternum/100)#, simulate.p.value = TRUE)
# this is the expected values if all proportions were the same
chisq.test(filternum)$expected
```

## Filter red-blood cells
This dataset has a few redbood cells, with high umi and low detected genes. 
We delete them using the ratio between these too parametres. (With the strict filtering it is not as essential, all these cells are probably gone eitherway)

```{r}
sce$ratio_detected_sum <- sce$detected / sce$sum
sce$outlier_ratio <- isOutlier(sce$ratio_detected_sum, type = "low")
plotColData(sce, x = "sum", y = "detected", colour_by = "outlier_ratio")
table(sce$outlier_ratio, sce$Sequencing)
```


## Perform the cell filtering

For the filtering I will use the thresholds used for the biological analysis of this data, keeping only 
the highest quality cells. 

```{r}
# perform filtering
sce <- sce[,!sce$filter & sce$outlier_ratio == FALSE]
# also filter genes
# empty genes
discard <- rowSums(assay(sce)) < 1 
sce <- sce[!discard, ]
```

## Gene QC

All transcripts that don't have at least 10 cells expressing it is discarded.We perform this 
QC separate for MGI and Illumina. 
The cell QC technically is separate too, it is per cell, and cells from MGI and cells from ILLUMINA are different. But the gene counts are calculated summing all cells, so we need to make sure to divide this into two datasets at this stage.
```{r}
#MGI  
sce_mgi <- sce[,sce$Sequencing == "MGI"]
sce_mgi <- addPerFeatureQCMetrics(sce_mgi)
rowData(sce_mgi)$gene_sum <- rowSums(counts(sce_mgi))
  
#Illumina
sce_illumina <- sce[,sce$Sequencing == "Illumina"]
sce_illumina <- addPerFeatureQCMetrics(sce_illumina)
rowData(sce_illumina)$gene_sum <- rowSums(counts(sce_illumina))

# Combine the rowdata from both technologies
#add suffix (only to the stuff added by featureQCmetrics)
colnames(rowData(sce_mgi))[4:6] <- paste0(colnames(rowData(sce_mgi))[4:6], "_mgi")
colnames(rowData(sce_illumina))[4:6] <- paste0(colnames(rowData(sce_illumina))[4:6], "_Illumina")


#mergee and add to original (from the mgi I add the whole thing, but first 3 cols are the same, so not added again for illuina)
rowData(sce) <-  cbind(rowData(sce_mgi), rowData(sce_illumina)[4:6])

## add some info for plotting
# average between the two technologies, if I want to plot genes with one single value 
rowData(sce)$mean <- rowMeans(as.matrix(rowData(sce)[,c("mean_mgi", "mean_Illumina")]))
rowData(sce)$detected <- rowMeans(as.matrix(rowData(sce)[,c("detected_mgi", "detected_Illumina")]))
# info what is found where
rowData(sce)$Only_in_one_sequencing_technology  <-  case_when(
  rowData(sce)$mean_mgi == 0 ~ "Only_Illumina",
  rowData(sce)$mean_Illumina == 0 ~ "Only_MGI",
  .default = "both")
# the counts is tricky, I don't want to inflate them by summing them when it's both
# nor reduce them by averaging when it's only in one.
rowData(sce)$gene_sum  <-  case_when(
  rowData(sce)$Only_in_one_sequencing_technology == "Only_Illumina" ~ rowData(sce)$gene_sum_Illumina,
  rowData(sce)$Only_in_one_sequencing_technology == "Only_MGI" ~ rowData(sce)$gene_sum_mgi,
  .default = rowMeans(as.matrix(rowData(sce)[,c("gene_sum_mgi", "gene_sum_Illumina")])))
  
# Discard if lower than 10.
rowData(sce)$discard <- ifelse(rowData(sce)$gene_sum > 10 , FALSE, TRUE)

# this need to wait after normalisation
# head(scuttle::makePerFeatureDF(sce))

# Plot different variables
ggplot(as.data.frame(rowData(sce)), aes(x=Only_in_one_sequencing_technology, y=mean)) +
  geom_violin() +
  geom_jitter(size = 0.1) +
  scale_y_log10(limits = c(min(rowData(sce)$mean), 100))#limits = c(0, 100)) # I delete the MALAT1

ggplot(as.data.frame(rowData(sce)), aes(x=Only_in_one_sequencing_technology, y=detected)) +
  geom_violin() +
    geom_jitter(size = 0.1) +
  scale_y_log10() +
  ylab("% cells where the gene is detected")

ggplot(as.data.frame(rowData(sce)), aes(x=Only_in_one_sequencing_technology, y=gene_sum)) +
 geom_violin(colour = "darkgrey", fill="white") +
  ggbeeswarm::geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, alpha=0.4, aes(colour = discard, y=gene_sum)) +
 #   geom_jitter(size = 0.1) +#, aes(colour = discard, y=gene_sum)) +
 # scale_y_log10() +
  scale_y_continuous(limits=c(0,3000000)) +
  scale_colour_manual(values = grey_red) +
  ylab("number cells where the gene is detected") +
  theme_classic()
```


```{r violin_only_one_seq_genes, fig.path=save_path}
ggplot(as.data.frame(rowData(sce)), aes(x=Only_in_one_sequencing_technology, y=gene_sum)) +
 geom_violin(colour = "darkgrey", fill="white") +
  ggbeeswarm::geom_quasirandom(width=0.4, groupOnX=TRUE, bandwidth=1, alpha=0.4, aes(colour = discard, y=gene_sum)) +
 #   geom_jitter(size = 0.1) +#, aes(colour = discard, y=gene_sum)) +
  scale_y_log10() +
  scale_colour_manual(values = grey_red) +
  ylab("number cells") +
  theme_classic()
```



# Normalisation, Variance estimation and Dim reduction

```{r dimred, fig.path=save_path}
library(scran)
if (!(file.exists(here("outs", "singlecell-workflow",  "sce_dim.RDS")))) {
#Normalisation
  # For reproducibility
  set.seed(100)
  # Quick clustering to pool samples together and deal with 0 counts
  quick_clusters <- quickCluster(sce)
  # Calculate size factors
  sce <- computeSumFactors(sce, cluster = quick_clusters, min.mean = 0.1)
  # Check that there are not negative size factors
  summary(sizeFactors(sce))
  # Apply size factors and log transform them
  sce <- logNormCounts(sce)

#Gene Variance
  gene_var_df <- modelGeneVar(sce)
  hvgs <- getTopHVGs(gene_var_df, prop=0.15)
  # save them in the object
  rowSubset(sce) <- hvgs
  
# PCA
  set.seed(1000)
  sce <- runPCA(sce)
  reducedDim(sce, "PCA") <- reducedDim(sce, "PCA")[,1:25]
# TSNE
  set.seed(1000)
  sce <- runTSNE(sce, dimred="PCA")

# Save
  saveRDS(sce,here("outs", "singlecell-workflow",  "sce_dim.RDS") )
}else{
  sce <- readRDS(here("outs", "singlecell-workflow",  "sce_dim.RDS"))
}
#I want to get both tanglled. Not MGI on top.
sce$random_order <- sample(ncol(sce))
# I first thought the colour_by field decided the order, so I extracted this into geom point
# maybe not. 
 # plotReducedDim(sce[,sce$random_order], dimred="TSNE", other_fields=c("Sequencing", "random_order")) + geom_point(aes(colour = Sequencing))
# With resampling sce in random order is enough
    plotReducedDim(sce[,sce$random_order], dimred="TSNE", colour_by = "Sequencing") + 
    scale_colour_manual(values=rev(blue_yellow))
  plotReducedDim(sce, dimred="TSNE", other_fields = "Sequencing", colour_by = "Sequencing") + facet_wrap(~Sequencing) + 
    scale_colour_manual(values=rev(blue_yellow))
  

 
```

# Annotation
```{r annotation, fig.path=save_path}
if(!(file.exists(here("outs", "singlecell-workflow",  "sce_anno.RDS")))) {
library(SingleR)
reference <- readRDS(here("data/singlecell-workflow/sce_th_annotated.RDS"))
annotation <- SingleR(test = sce, ref = reference, labels = reference$celltype, de.method = "wilcox")
sce$labels <- annotation$labels
sce$labels_pruned <- annotation$pruned.labels
sce$clusters_named <- stringr::str_remove(sce$labels_pruned, "^th")
saveRDS(sce, here("outs", "singlecell-workflow",  "sce_anno.RDS"))
sce <- sce[,(!is.na(sce$labels_pruned))]
saveRDS(sce, here("outs", "singlecell-workflow",  "sce_anno_noNA.RDS"))
}else{
  sce <- readRDS(here("outs", "singlecell-workflow",  "sce_anno_noNA.RDS"))
}
plotReducedDim(sce, dimred="TSNE", other_fields = "Sequencing", colour_by = "clusters_named") + facet_wrap(~Sequencing) + scale_colour_manual(values = kelly_col)
```


There is no significant difference between the two Sequencing methods in the annotated number of cells. 

```{r}
chisq.test(table(sce$Sequencing, sce$clusters_named))
```

```{r bar_plot_annotation_prop, fig.path=save_path}
prop <- as.data.frame(table(sce$Sequencing, sce$clusters_named))
colnames(prop) <- c("Sequencing", "CellType", "Proportion")
ggplot(data = prop, aes(x = CellType , y = Proportion, fill = Sequencing)) +
  geom_bar(position = "fill", stat = "identity") + theme_classic()+
  scale_fill_manual(values = rev(blue_yellow)) +
theme(axis.text.x = element_text(angle = 45, hjust=1, vjust = 1)) +
  labs( x = element_blank(), 
        y = element_text("Proportion"))
```


```{r bar_plot_annotation_nums, fig.path=save_path}
ggplot(data = prop, aes(x = CellType , y = Proportion, fill = Sequencing)) +
  geom_bar(stat = "identity") + theme_classic()+
  scale_fill_manual(values = rev(blue_yellow)) +
theme(axis.text.x = element_text(angle = 45, hjust=1, vjust = 1)) +
  labs( x = element_blank(), 
        y = element_text("Cell Numbers"))

table(sce$Sequencing, sce$labels_pruned)
```

